src/main/kotlin/com/skaggsm/treechoppermod/LogBlockUtils.kt

```diff
@@ -58,6 +59,8 @@ private val directions = linkedSetOf(
     // Reversed so that the top gets added to the output list last and gets picked first. Makes log breaking look more "natural".
     .reversed()
 
+private val WAS_TOUCHING_NATURAL_LEAVES = CacheBuilder.newBuilder().maximumSize(1024L * 64).build<BlockPos, Unit>()
```

Comment:
This cache should take up <2MiB at _max_, so I'm not worried about it getting too large